CPB_AMS_Gizmo_BLE/cpb_ams_gizmo_ble.py

```diff
@@ -0,0 +1,130 @@
+"""
+This example solicits that apple devices that provide notifications connect to it, initiates
+pairing, prints existing notifications and then prints any new ones as they arrive.
+"""
+
+import time
+import displayio
+import terminalio
+from adafruit_gizmo import tft_gizmo
+from adafruit_display_text.label import Label
+from adafruit_display_shapes.rect import Rect
+from adafruit_bitmap_font import bitmap_font
+import adafruit_ble
+from adafruit_ble.advertising.standard import SolicitServicesAdvertisement
+from adafruit_ble_apple_media import AppleMediaService
+from adafruit_circuitplayground import cp
+
+BACKGROUND_COLOR = 0x444444 # Gray
+TEXT_COLOR = 0xFF0000 # Red
+BORDER_COLOR = 0xAAAAAA # Light Gray
+STATUS_COLOR = BORDER_COLOR
+
+# PyLint can't find BLERadio for some reason so special case it here.
+radio = adafruit_ble.BLERadio() # pylint: disable=no-member
+a = SolicitServicesAdvertisement()
+a.solicited_services.append(AppleMediaService)
+radio.start_advertising(a)
+
+def wrap_in_tilegrid(open_file):
+    odb = displayio.OnDiskBitmap(open_file)
+    return displayio.TileGrid(odb, pixel_shader=displayio.ColorConverter())
+
+def make_background(width, height, color):
+    color_bitmap = displayio.Bitmap(width, height, 1)
+    color_palette = displayio.Palette(1)
+    color_palette[0] = color
+
+    return displayio.TileGrid(color_bitmap,
+                              pixel_shader=color_palette,
+                              x=0, y=0)
+
+def load_font(fontname, text):
+    font = bitmap_font.load_font(fontname)
+    font.load_glyphs(text.encode('utf-8'))
+    return font
+
+def make_label(text, x, y, color, max_glyphs=30, font=terminalio.FONT):
+    if isinstance(font, str):
+        font = load_font(font, "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz.,?()")
+    text_area = Label(font, text=text, color=color, max_glyphs=max_glyphs)
+    text_area.x = x
+    text_area.y = y
+    return text_area
+
+def set_label(label, value, max_length):
+    text = "{}".format(value)
+    if len(text) > max_length:
+        text = text[:max_length-3] + "..."
+    label.text = text
+
+def set_status(label, action_text, player):
+    label.text = "{} on {}".format(action_text, player)
+    _, _, label_width, _ = label.bounding_box
+    label.x = display.width - 10 - label_width
+
+display = tft_gizmo.TFT_Gizmo()
+group = displayio.Group(max_size=20)
+display.show(group)
+
+while True:
+    if not radio.connected:
+        group.append(wrap_in_tilegrid(open("/graphic_tfts_ams_connect.bmp", "rb")))
+
+        while not radio.connected:
+            pass
+
+        group.pop()
+    print("connected")
+
+    known_notifications = set()
+
+    # Draw the text fields
+    print("Loading Font Glyphs...")
+    title_label = make_label("None", 18, 30, TEXT_COLOR, font="/fonts/Arial-Bold-24.bdf")
```

Comment:
this can move left since we'll scale the rectangle up a bit, maybe 12 instead of 18, although this may change if we go with a smaller font.

Reply:
Any suggestions on a font size? I was trying to find a balance between it showing lots of text and making it large like in your mockup. Like maybe 18 instead of 22?

Reply:
I went with 18/16 and 12 is still fine.

Reply:
thanks, I'll check it out.

Reply:
the font sizes look good! I'm seeing 16 and bold 18, but not sure what you are referring to with "12 is still fine"?

Reply:
oh, sorry, I see, it's the position not the font size, nevermind :)